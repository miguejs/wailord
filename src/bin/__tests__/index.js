import path from 'path'
import spawn from 'spawn-command'
import stripAnsi from 'strip-ansi'

const SPLIT_GUIDE_PATH = require.resolve('../index')
const BABEL_BIN_PATH = require.resolve('babel-cli/bin/babel-node')

test('wailord --help', () => {
  return runWailordGuideCLI('--help').then(stdout => {
    expect(stdout).toMatchSnapshot('wailord --help stdout')
  })
})

function runWailordGuideCLI(args = '', cwd = process.cwd()) {
  const isRelative = cwd[0] !== '/'
  if (isRelative) {
    cwd = path.resolve(__dirname, cwd)
  }

  return new Promise((resolve, reject) => {
    let stdout = ''
    let stderr = ''
    const command = `${BABEL_BIN_PATH} -- ${SPLIT_GUIDE_PATH} ${args}`
    const child = spawn(command, {cwd})

    child.on('error', error => {
      reject(error)
    })

    child.stdout.on('data', data => {
      stdout += stripAnsi(data.toString())
    })

    child.stderr.on('data', data => {
      stderr += stripAnsi(data.toString())
    })

    child.on('close', () => {
      if (stderr) {
        reject(stderr)
      } else {
        resolve(stdout)
      }
    })
  })
}
