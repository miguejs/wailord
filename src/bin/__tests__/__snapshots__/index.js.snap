// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wailord --help: wailord --help stdout 1`] = `
"
  Usage: index [options] [command]

  Options:

    -V, --version  output the version number
    -h, --help     output usage information

  Commands:

    attach <req>   Attach local standard input, output, and error streams to a running container
"
`;
